packages/core/src/initialization/make-index-api.ts

```diff
@@ -27,13 +11,26 @@ export function makeIndexApi(
   network: Networks,
   logger: DiagnosticsLogger
 ): DatabaseIndexApi | undefined {
-  if (process.env.CERAMIC_ENABLE_EXPERIMENTAL_COMPOSE_DB != 'true') {
-    return undefined
-  }
   if (!indexingConfig) {
     logger.warn(`Indexing is not configured. Please add the indexing settings to your config file`)
     return undefined
   }
+
+  if (indexingConfig.composedbEnabled === false) {
+    logger.warn(
+      'Indexing is actively disabled. Change the corresponding CLI flag or ENV variable to enable it.'
+    )
+    return undefined
+  }
+
+  // TODO: extend with dynamic config check if MAINNET or sync is enabled
```

Comment:
what does "dynamic config check" mean?  Also can we link a linear ticket for the TODO?